src/Domain.Tests/TicTacToeGameTests.cs

```diff
@@ -0,0 +1,508 @@
+                Player = Player.Cross,
+                BoardPosition = new Position(2, 0),
+                TilePosition = new Position(2, 0),
+                MoveNumber = 2
+            });
+
+            // assert
+            result.IsValid.Should().BeFalse();
+            result.InvalidReason.Should().Be(ExceptionMessages.TileNotEmpty);
+        }
+
+        [Fact]
+        public void Move_MoveOnWrongBoard_InvalidMoveInvalidPosition()
+        {
+            // arrange
+            var game = new TicTacToeGame(new[]
+            {
+                new Move
+                {
+                    Player = Player.Cross,
+                    BoardPosition = new Position(0, 0),
+                    TilePosition = new Position(2, 0),
+                    MoveNumber = 1
+                }
+            });
+
+            // act
+            var result = game.Move(new Move
+            {
+                Player = Player.Circle,
+                BoardPosition = new Position(2, 1),
+                TilePosition = new Position(2, 0),
+                MoveNumber = 2
+            });
+
+            // assert
+            result.IsValid.Should().BeFalse();
+            result.InvalidReason.Should().Be(ExceptionMessages.IllegalPosition);
+        }
+
+        [Fact]
+        public void Move_MoveOnValidButFullBoard_InvalidMoveInvalidPosition()
+        {
+            // arrange
+            var game = new TicTacToeGame(new[]
+            {
+                new Move
+                {
+                    Player = Player.Cross,
+                    BoardPosition = new Position(0, 0),
+                    TilePosition = new Position(2, 0),
+                    MoveNumber = 1
+                },
+                new Move
+                {
+                    Player = Player.Circle,
+                    BoardPosition = new Position(2, 0),
+                    TilePosition = new Position(0, 0),
+                    MoveNumber = 2
+                },
+                new Move
+                {
+                    Player = Player.Cross,
+                    BoardPosition = new Position(0, 0),
+                    TilePosition = new Position(1, 0),
+                    MoveNumber = 3
+                },
+                new Move
+                {
+                    Player = Player.Circle,
+                    BoardPosition = new Position(1, 0),
+                    TilePosition = new Position(0, 0),
+                    MoveNumber = 4
+                },
+                new Move
+                {
+                    Player = Player.Cross,
+                    BoardPosition = new Position(0, 0),
+                    TilePosition = new Position(0, 0),
+                    MoveNumber = 5
+                },
+            });
+
+            // act
+            var result = game.Move(new Move
+            {
+                Player = Player.Circle,
+                BoardPosition = new Position(0, 0),
+                TilePosition = new Position(0, 0),
```

Comment:
I think this position should be on an empty tile for testing exactly what you want. 
Like: 1,1 or 2,2.
This tile was filled with Move 5.

Reply:
good point. changed it